crates/rpc/rpc/src/debug.rs

```diff
@@ -133,10 +134,32 @@ where
             Block::decode(&mut rlp_block.as_ref()).map_err(BlockError::RlpDecodeRawBlock)?;
 
         let (cfg, block_env) = self.inner.eth_api.evm_env_for_raw_block(&block.header).await?;
-
         // we trace on top the block's parent block
         let parent = block.parent_hash;
-        self.trace_block_with(parent.into(), block.body, cfg, block_env, opts).await
+
+        // Depending on EIP-2 we need to recover the transactions differently
+        let transactions =
+            if self.inner.provider.chain_spec().is_homestead_active_at_block(block.number) {
+                block
+                    .body
+                    .into_iter()
+                    .map(|tx| {
+                        tx.into_ecrecovered()
+                            .ok_or_else(|| EthApiError::InvalidTransactionSignature)
+                    })
+                    .collect::<EthResult<Vec<_>>>()?
+            } else {
+                block
+                    .body
+                    .into_iter()
+                    .map(|tx| {
+                        tx.into_ecrecovered_unchecked()
+                            .ok_or_else(|| EthApiError::InvalidTransactionSignature)
+                    })
+                    .collect::<EthResult<Vec<_>>>()?
+            };
```

Comment:
should we make it a util fn/method?

Reply:
that's really the only location where we need this,
but can create a good first issue for this